### PR TITLE
[ADD] udes_common: Helper function selection_value_displayname()

### PR DESCRIPTION
**Pull request by PeterAlabaster**
Does what it says on the tin, gets the displayname for a given
selection fields value on a recordset

Also added a readme and filled in what i could for what already exists

Signed-off-by: Peter Alabaster <peter.alabaster@unipart.io>

User-story: 411

### CODE REVIEW

File: addons/udes_common/tests/test_selection_display_name.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        """Ensure function returns False if the selection field on the recordset is not set"""
+        self.model_overview_report.report_type = False
+
+        normal_output = self.model_overview_report.report_type
+        self.assertEqual(normal_output, False)
+
+        alt_output = self.model_overview_report.selection_value_displayname("report_type")
+        self.assertEqual(alt_output, False)
+
+    # NB: It's not possible (as far as i'm aware) to mock a new field on a model without lots of pain
```
Comment (PeterAlabaster) [inline addons/udes_common/tests/test_selection_display_name.py:105]: Left this in incase anyone has ideas

Comment (PeterAlabaster) [reply addons/udes_common/tests/test_selection_display_name.py]: Removed this as there is really no need to test it as we know it works and it's too tricky to mock

**Review by armandcela — APPROVED** [review]

**Review by adamrpatrick — APPROVED** [review]

File: addons/udes_common/models/models.py [hunk 2]
```diff
@@ -84,3 +85,23 @@ def statistics(self, cache=False):
 def trace(self, filter=None, max=None):
     """Trace database queries"""
     return tools.QueryTracer(self.env.cr, filter=filter, max=max)
+
+
+@add_if_not_exists(models.BaseModel)
+def selection_value_displayname(self, selection_field_name):
```
Comment (PeterC10) [inline addons/udes_common/models/models.py:91]: Super nit (well two): I think we can shorten this to `selection_display_name`

1. Don't think value is needed
2. Think `display_name` is tidier than `displayname`

File: addons/udes_common/models/models.py [hunk 3]
```diff
@@ -84,3 +85,23 @@ def statistics(self, cache=False):
...
+    self.ensure_one()
+    try:
+        selection_field = self._fields[selection_field_name]
+    except KeyError:
+        raise ValidationError(
+            _(f"Field '{selection_field_name}' does not exist on model '{self._name}'")
+        )
+    # Odoo has a nice function that handles all the weird ways you can define selections!
+    selection_values_dict = dict(selection_field._description_selection(self.env))
+    return selection_values_dict.get(self[selection_field_name], False)
```
Comment (PeterC10) [inline addons/udes_common/models/models.py:107]: Not sure if I'm misunderstanding something but if we can't find a display name for the value I think it is better to return the code value rather than `False`.

File: addons/udes_common/tests/test_selection_display_name.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+        # A random recordset on a model with a normal selection field
+        cls.model_overview_report = cls.env.ref("base.report_ir_model_overview")
+        # A random recordset on a model with a generated selection field (function)
+        cls.main_partner = cls.env.ref("base.main_partner")
+        # There are no recordsets on res.partner.bank, but it has a lambda generated
+        # selection field, so we just create one
+        cls.res_partner_bank_recordset = ResPartnerBank.create(
+            dict(
+                acc_number="60-16-13 31926819",
```
Comment (PeterC10) [inline addons/udes_common/tests/test_selection_display_name.py:35]: Nit: Leaking my bank details online... :D

Comment (PeterAlabaster) [reply addons/udes_common/tests/test_selection_display_name.py:35]: Taken from odoo/odoo/addons/base/data/res_partner_bank_demo.xml

File: addons/udes_common/models/models.py (hunk 3 above)
Comment (PeterAlabaster) [reply addons/udes_common/models/models.py:107]: This is more so if the selection fields value is set to False, the .get() will return None.
Just trying to keep consistency with how Odoo behaves when getting the normal selection field value

**Review by PeterC10 — APPROVED** [review]
